snowplow_tracker/tracker.py

```diff
@@ -1014,7 +1020,7 @@ def flush(self, is_async: bool = False) -> "Tracker":
                     emitter.sync_flush()
         return self
 
-    def set_subject(self, subject: Optional[Subject]) -> "Tracker":
+    def set_subject(self, subject: Subject) -> "Tracker":
```

Comment:
I wonder if we should keep the `Optional` here? Looking at how the `subject` is used, it seems that we (passed to `event.build_payload()`, it seems that it can be optional?